packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -124,15 +127,35 @@ export function createOdspNetworkError(
     props: ITelemetryProperties = {},
 ): LoggingError & OdspError & IFacetCodes {
     let error: LoggingError & OdspError & IFacetCodes;
+    const parseResult = tryParseErrorResponse(responseText);
+    let facetCodes: string[] | undefined;
+    let innerMostErrorCode: string | undefined;
+    if (parseResult.success) {
+        // Log the whole response if it looks like the error format we expect
+        props.response = responseText;
+        const errorResponse = parseResult.errorResponse;
+        facetCodes = parseFacetCodes(errorResponse);
+        if (facetCodes !== undefined) {
+            innerMostErrorCode = facetCodes[0];
+            props.innerMostErrorCode = innerMostErrorCode;
+        }
+    }
     switch (statusCode) {
         case 400:
             error = new GenericNetworkError(errorMessage, false, { statusCode });
             break;
         case 401:
         case 403:
-            const claims = response?.headers ? parseAuthErrorClaims(response.headers) : undefined;
-            const tenantId = response?.headers ? parseAuthErrorTenant(response.headers) : undefined;
-            error = new AuthorizationError(errorMessage, claims, tenantId, { statusCode });
+            if (innerMostErrorCode === OdspServiceReadOnlyErrorCode) {
```

Comment:
Maybe add some comment here on why we differentiate this case and how it's handled in the code base, to help next engineer :)